com.civicrm.hrjobroles/tests/phpunit/HrJobRolesTestBase.php

```diff
@@ -98,6 +108,27 @@ public function createSampleOptionGroupsAndValues()  {
       $params = ['option_group_id' => $groupID, 'name' => $value, 'value' => $value ];
       CRM_Core_BAO_OptionValue::create($params);
     }
+
+  }
+
+  /**
+   * Find and retrieve job role by any of its properties
+   *
+   * @param int|string $value
+   * @param string $identifier
+   *
+   * @return \CRM_Hrjobroles_BAO_HrJobRoles|NULL
+   */
+  public function findRoleByProperty($value, $identifier = 'id')  {
```

Comment:
Oh, it's must have been introduced recently https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/DAO.php#L988
But you can still use commonRetrieve and pass the id in the $params array.